examples/prom/maxwell_local_rom_greedy.cpp

```diff
@@ -0,0 +1,737 @@
+        double curr_error = 0;
+
+        // 29a. Calculate the relative error as commanded by the greedy algorithm.
+        if (calc_rel_error)
+        {
+            Vector true_solution(X.Size());
+            ifstream solution_file;
+            std::string solution_filename = "Sol" + curr_basis_identifier + to_string(myid);
+            if (myid == 0) cout << "Comparing current run to solution at: "
+                                    << solution_filename << endl;
+            solution_file.open(solution_filename);
+            true_solution.Load(solution_file, X.Size());
+            solution_file.close();
+            Vector residual(X.Size());
+            subtract(X, true_solution, residual);
+
+            const double abs_error = sqrt(InnerProduct(MPI_COMM_WORLD, residual, residual));
+            const double sol_norm = sqrt(InnerProduct(MPI_COMM_WORLD, true_solution,
+                                         true_solution));
+
+            curr_error = abs_error / sol_norm;
+            if (myid == 0) cout << "The relative error is: " << curr_error << endl;
+
+            greedy_sampler->setPointRelativeError(curr_error);
+        }
+        // 29b. Or calculate the error indicator as commanded by the greedy algorithm.
+        else if (calc_err_indicator)
+        {
+            Vector AX(X.Size());
+//            A.Mult(X, AX);
+            A.As<HypreParMatrix>()->Mult(X, AX);
+            Vector residual(X.Size());
+            subtract(B, AX, residual);
+
+            curr_error = sqrt(InnerProduct(MPI_COMM_WORLD, residual, residual));
+            if (myid == 0) cout << "The error indicator is: "
+                                    << curr_error << endl;
+            greedy_sampler->setPointErrorIndicator(curr_error, 1);
+        }
+        // 29c. Or if not using the greedy algorithm, output the relative error
+        //     the error indicator of the point. (just for extra information)
+        else if (!build_database && online)
+        {
+            Vector true_solution(X.Size());
+            ifstream solution_file;
+            std::string solution_filename = "Sol" + curr_basis_identifier + to_string(myid);
+            if (myid == 0) cout << "Comparing current run to solution at: "
+                                    << solution_filename << endl;
+            solution_file.open(solution_filename);
+            true_solution.Load(solution_file, X.Size());
+            solution_file.close();
+            Vector residual(X.Size());
+            subtract(X, true_solution, residual);
+
+            const double abs_error = sqrt(InnerProduct(MPI_COMM_WORLD, residual, residual));
+            const double sol_norm = sqrt(InnerProduct(MPI_COMM_WORLD, true_solution,
+                                         true_solution));
+
+            curr_error = abs_error / sol_norm;
+            if (myid == 0) cout << "The relative error is: " << curr_error << endl;
+
+            Vector AX(X.Size());
+//            A.Mult(X, AX);
```

Comment:
This line can be removed.

Reply:
Thanks! just deleted all of the extra lines/variables you pointed out.